platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/utility/RecycleBinTests.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.test.utility;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.swirlds.common.io.utility.FileUtils;
+import com.swirlds.common.io.utility.RecycleBin;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.test.framework.config.TestConfigBuilder;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+@DisplayName("RecycleBin Tests")
+class RecycleBinTests {
+
+    /**
+     * Temporary directory provided by JUnit
+     */
+    @TempDir
+    Path testDirectory;
+
+    @BeforeEach
+    void beforeEach() throws IOException {
+        FileUtils.deleteDirectory(testDirectory);
+    }
+
+    @AfterEach
+    void afterEach() throws IOException {
+        FileUtils.deleteDirectory(testDirectory);
+    }
+
+    /**
+     * Create a file and write a string to it.
+     */
+    private void writeFile(@NonNull final Path path, @NonNull final String contents) throws IOException {
+        final Path parent = path.getParent();
+        Files.createDirectories(parent);
+
+        final BufferedWriter writer = Files.newBufferedWriter(path);
+        writer.write(contents);
+        writer.close();
+    }
+
+    /**
+     * Validate that a file exists and contains the expected contents.
+     */
+    private void validateFile(@NonNull final Path path, @NonNull final String expectedContents) throws IOException {
+        assertTrue(Files.exists(path));
+        final String contents = Files.readString(path);
+        assertTrue(contents.equals(expectedContents));
+    }
+
+    @Test
+    @DisplayName("Recycle File Test")
+    void recycleFileTest() throws IOException {
+        final Configuration configuration = new TestConfigBuilder()
```

Comment:
nit: this block is duplicated in all tests and could be pulled into the `beforeEach`

```
        final Configuration configuration = new TestConfigBuilder()
                .withValue("state.savedStateDirectory", testDirectory.toString())
                .getOrCreateConfig();

        final RecycleBin recycleBin = new RecycleBin(configuration, new NodeId(0));
```

Reply:
I made it so that the configuration is setup in the `beforeEach()`. I'd prefer not to set up the `RecycleBin` object there though, I like to be able to see the entire lifecycle of the object I'm testing in the unit test function.